crates/iceberg/src/scan.rs

```diff
@@ -302,13 +262,147 @@ impl TableScan {
 
         arrow_reader_builder.build().read(self.plan_files().await?)
     }
+}
+
+#[derive(Debug)]
+struct ConcurrentFileScanStreamContext {
```

Comment:
I think once we have a runtime and the async approach is approved, we can get rid of the `FileScanStreamContext` and merge the struct into `ConcurrentFileScanStreamContext`

Reply:
Yeah, completely agree. Kept them separate for now as it made it easier to have the old non-concurrent version in the code base at the same time.

Reply:
I've refactored these into a single context. Now that we are using the `ExpressionEvaluator` though, I ended up creating another context object that gets passed to the manifest file closure. This was just a tuple before but clippy complained when I added one more item to it :-)